### ISSUE TITLE
Error opening .sh files using 'Execute in Terminal' (Valid Key file could not be found in search dirs)

### ISSUE THREAD

**Issue by Bluey26** [lxqt/pcmanfm-qt#1902]
##### Expected Behavior

Opening a .sh file using the file-dialog option should open qterminal or any other set terminal with the .sh file associated. 

##### Current Behavior

An error is shown when i try 'Execute in Terminal' using the pcmanfm-qt file dialog screen. (see screens)
##### Possible Solution

Could it be related to the WIP qt6 branch of qterminal? I have installed `https://github.com/qterminal#branch=use_layershell_dropdown` 

##### Steps to Reproduce (for bugs)

1.Open pcmanfm-qt or a .sh file present in the desktop screen
2.Select 'Execute in Terminal' option
3.The error dialog appears

##### Context

Updated to LXQt 2.0 today, using LabWC lxqt session AURs, and the WIP lxqt-panel and qterminal branches found in 
https://github.com/stefonarch/LXQt-Wayland-files/blob/main/AUR/

##### System Information

* PcManFm-qt version: 2.0
* Distribution & Version: Archlinux last updated= today
* Qt Version: 6.7.0
* liblxqt Version: 2.0.0-1

![screen_Mon_1modif](https://github.com/lxqt/pcmanfm-qt/assets/143142826/53de0cd0-e0fc-4f1e-997c-499e9f7bb828)

![screen_Mon_2m](https://github.com/lxqt/pcmanfm-qt/assets/143142826/191e5df7-d5ec-4ad4-a4d2-96ec96771ed6)

---

Other files open normally (images,videos,etc). The same .sh file can be 'Executed' normally (the other option present in the dialog)

PS: using F4(opens the qterminal inside the folder) and using `./screenshot.sh` works properly.

**Comment by tsujan** [comment]
Not reproducible.

> Could it be related to the WIP qt6 branch of qterminal?

If it happens with other terminals, no.

> Updated to LXQt 2.0 today

Ssomething may have gone wrong with the update. Incomplete installation(s)?

> using LabWC lxqt session AURs,

Please use a standard X11 session and report back.

**Comment by Bluey26** [comment]
Same issue happens to me in a X11 session ( 'Valid key file could not be found in search dirs') when i try to open a .sh file using that option.

Yes, you are right, it happens with other terminals set in pcmanfm-qt preferences, so it could not be qterminal.

Another used have reported it a few hours ago in the matrix LXQt channel, same error.

I think i have installed everything form the 2.0. I removed the AURs before updating to the stable archlinux repositories, then i installed the mentioned AURs.

**Comment by tsujan** [comment]
> Another user have reported it a few hours ago in the matrix LXQt channel, same error.

Which distro? And what about others?

**Comment by Bluey26** [comment]
> > Another user have reported it a few hours ago in the matrix LXQt channel, same error.
> 
> Which distro? And what about others?

The other person seems to be using Archlinux too (I'm using Archlinux). I just have been testing things and 'open with' commands also have this problem. Those 'open with' files are generated inside ~/.local/share/applications.


I noticed that there was a new 'flag' inside, which does not seem to do anything with this issue, but if i change `Terminal=true` to `Terminal=false` , the open with option works sucessfuly(it does not open a terminal tab, is executed in the background).

This only affects those 'open with' options which needs or involve a terminal(those with `Terminal=true` inside the .desktop)

**Comment by tsujan** [comment]
> but if i change Terminal=true to Terminal=false , the open with option works sucessfuly

Not reproducible again. Actually, nothing has changed in libfm-qt or pcmanfm-qt in this regard.

We might need to hear from other Arch users too. Although I have Manjaro (= 99% Arch), I use my own LXQt packages. If  other Arch users have your problem, that might indicate there's a problem in Arch; if not, the problem is local.

**Comment by tsujan** [comment]
> Those 'open with' files are generated inside ~/.local/share/applications.

What's the output of `echo $XDG_DATA_HOME` for you?

**Comment by Bluey26** [comment]
`echo $XDG_DATA_HOME` = `/home/MY_USER/.local/share`

`echo $XDG_DATA_DIRS` = `/home/MY_USER/.local/share:/usr/local/share:/usr/share`

**Comment by tsujan** [comment]
OK, I found the problem: It is a bug in libfm-qt. Will make a PR soon.

The reason that I didn't see it was just that I've also the Qt5-based libfm-qt (which can coexist with the Qt6 version and provide LXQt file dialog for Qt5 apps). Once I renamed `/usr/share/libfm-qt/terminals.list`, I could reproduce the issue.

**Comment by Bluey26** [comment]
Excellent, thank you very much!!

**Comment by tsujan** [comment]
@Bluey26 
The fix is here: https://github.com/lxqt/libfm-qt/pull/972
Your test would be very appreciated. Thank you so much for your good report and fast response!

**Comment by Bluey26** [comment]
Thank to you for the fast fix!

I confirm it works (Now 'Execute in Terminal' works and there is no error popup).

To install the fix, i have to use the libfm-qt PKGINFO/SRCBUILD of the arch repositories(not git one) and set ` IgnorePkg   = libfm-qt` in the meanwhile to avoid the system to re-install the master branch.

I comment this because if somebody wants to try this, they will find that using the `libfm-qt-git` package would not work because other qt programs will 'complain'(because they will ask for  `libfm-qt` and not for `libfm-qt-git`.

So, for other persons looking to apply this fix, i cloned:
https://gitlab.archlinux.org/archlinux/packaging/packages/libfm-qt
(`git clone https://gitlab.archlinux.org/archlinux/packaging/packages/libfm-qt.git`)

Then in the `PKGINFO` i added the following(and replaced the original lines):

```
source=('git+https://github.com/lxqt/libfm-qt#branch=fix_terminal_list')
sha256sums=('SKIP')

build() {
	cmake -B build -S "$srcdir/$pkgname" \
		-DCMAKE_INSTALL_PREFIX=/usr \
		-DCMAKE_BUILD_TYPE=None
	make -C build
}
```
The Build part is only because `cmake -B build -S "$srcdir/$pkgname-$pkgver"` would just not work(the git branch creates `$srcdir/$pkgname`

In the other file, `.SRCINFO` i have changed (and replaced the original lines):

```
	source = git+https://github.com/lxqt/libfm-qt#branch=fix_terminal_list
	sha256sums = SKIP
```

After that and the PKGIGNORE flag in pacman, everything goes smooth and the package replaces the one in stable repos. After logout, the fix started to work and everything seems OK.

---
I dont want to rush you with issues since the stable packages left testing yesterday, but i have noticed 2 things, so i mention them in here in case nobody is aware of them already:

Featherpad does not focus when another file is already open in labwc(i think this is a known thing, can be solved by open every file in a separated window, which featherpad supports and have an option for)
Featherpad does not seem to detect the CTRL+SHIFT+R shortcut to reload a file. This happens when a file is modified(and a red message appears saying it). In the other hand, the menu option (File>Reload) seems to work properly.

I noticed that `qps` 'used' ram seems to not match what `free -m` says (this happens both in X11 an labwc-wayland session). The number shown by qps is lower than the one informed by `free -m`. I noticed this because my conky panel was showing a bigger used ram number, then i realized that the one shown i qps was strangely low.

As i said before, dont take this as a rush to fix the things, everything seems to work really fine, they are 'small details' i have been found and wanted to report back to you become aware of them (in case you did not already).

Thank you again for the fix and to the whole team for all the work done to bring this 2.0 LXQt Release.

**Comment by tsujan** [comment]
The fix will soon be released as libfm-qt 2.0.2 and will be in Arch.

> I don't want to rush you with issues...

On the contrary, we appreciate bug reports; they're a very important part of the development.

The rest of your comment wasn't about pcmanfm-qt, but since your main report was very important, I give short answers here ;) Please use [Discussions](https://github.com/lxqt/lxqt/discussions) for questions, ideas,...

> Featherpad does not focus when another file is already open in labwc

This is about Wayland: apps can neither activate their windows nor set their positions. When I'm under LXQt+LabWC (most of the time), I choose "Always open in separate windows".

> Featherpad does not seem to detect the CTRL+SHIFT+R shortcut to reload a file...

An old case of a shortcut clash: most probably, `CTRL+SHIFT+R` is taken by LabWC. The solution is changing one of them (preferably, FeatherPad's).

> I noticed that qps 'used' ram seems to not match what free -m says

Yes, I've noticed that too. What Qps, gnome-system-monitor and Conky show are mostly different. I think they use different interpretations of "usage". This can be reported at Qps for further investigation, but I vaguely remember that I checked it and was convinced that Qps wasn't wrong.

**Comment by Bluey26** [comment]
Thank you very much, it seems that the featherpad hotkey is already being used by labwc.
I will create the issue for qps just in case, and add more information.
